### PR TITLE
Added category directive and added category attribute to test case

### PR DESCRIPTION
**Pull request by derekonay**
Fix for Issue #176 , may need another test case, unsure of that.

### CODE REVIEW

File: lib/ui/analytics/tests/analytics-spec.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ describe('avAnalytics', function() {
   availity.mock.directiveSpecHelper();
 
   beforeEach(inject(function(_avAnalytics_, $q) {
-    $el = availity.mock.compileDirective('<button data-av-analytics-on="click" data-av-analytics-action="save"></button>');
+    $el = availity.mock.compileDirective('<button data-av-analytics-on="click" data-av-analytics-category="test category" data-av-analytics-label="save"></button>');
```
Comment (robmcguinness) [inline lib/ui/analytics/tests/analytics-spec.js]: why was the spec changed?

Comment (derekonay) [inline lib/ui/analytics/tests/analytics-spec.js]: I will change it back, came to the conclusion we need a test case for Splunk and a separate test case for Piwik. The changes I made here reflect that you get the correct results for a Piwik event.

**Comment by robmcguinness** [comment]
I believe we are trying to reduct the boilerplate required to wire up a group of analytic attributes for a set of dom elements. correct?  if so:

```html
<div av-analytics="{'category' : 'vidoes'}">
 <!-- child directives -->
</div>
```

or 

```html
<div av-analytics"myOptions">
 <!-- child directives -->
</div>
```

or 

```js
myApp.controller('MyCtrl', function ($scope) {
  $scope.myOptions = {category: "vidoes"}
});
```

and
```js
ui.directive('avAnalytics', function() {
  return {            
    link: function(...) {
      // get options here from from $scope 
    }    
  };
});
```

At this point we don't need to set any attributes on the child av-analytics-on directive (at least I don't think), just read the options from the parent controller.

**Comment by robmcguinness** [comment]
Here is my understanding: We are trying to reduce the boilerplate needed to configure a grouping of analtyic events.  So I propose the following:

* remove `av-analytic-category`: seems like this directive should exist because it's one of the possible options that a analytic event can have

```html
<div av-analytics="{'category' : 'vidoes'}">
 <!-- child directives -->
</div>
```

or 

```html
<div av-analytics av-options="myOptions">
 <!-- child directives -->
</div>
```

or 

```js
myApp.controller('MyCtrl', function ($scope) {
  $scope.myOptions = {category: "vidoes"}
});
```

and

```js
ui.directive('avAnalytics', function() {
  return {            
    link: function(...) {
      // get options here from $scope
    }    
  };
});
```

The `av-analytics-on` directive would be a child of the `av-analtyics` parent directive so just read the attributes from the parent to configure the child.  thoughts?

**Comment by derekonay** [comment]
I'd have to agree with you here. I just noticed my current changes will break any app that does not define a category directives. 

**Example:**  If you are wanting to log to Splunk, it will throw an error.

I'm going to try the first approach above.

**Pull request merged**